frontend/src/metabase/databases/components/DatabaseConnectionUri/DatabaseConnectionStringField.tsx

```diff
@@ -97,6 +101,20 @@ export function DatabaseConnectionStringField({
     setConnectionString,
   ]);
 
+  function handleBlur() {
+    match(lastClearedStatusRef.current)
+      .with("success", () => {
+        connectionStringParsedSuccess(location);
+      })
+      .with("failure", () => {
+        connectionStringParsedFailed(location);
+      })
+      .with(null, () => {
+        return;
+      })
+      .exhaustive();
```

Comment:
Nit:
I really don't see any advantage in using ts-pattern's `match` here as these conditionals are pretty simple and we aren't returning a value. 
Two ifs aren't a "nested if/else hell" and would be simpler to understand at a glance in this specific case.
```suggestion
    if (lastClearedStatusRef.current === "success") {
      connectionStringParsedSuccess(location);
    }
    if (lastClearedStatusRef.current === "failure") {
      connectionStringParsedFailed(location);
    }
```